cmd/sfncli/runner.go

```diff
@@ -20,36 +20,44 @@ const maxTaskOutputLength = 32768
 const maxTaskFailureCauseLength = 32768
 
 type TaskRunner struct {
-	sfnapi    sfniface.SFNAPI
-	taskToken string
-	cmd       string
-	args      []string
+	sfnapi        sfniface.SFNAPI
+	taskToken     string
+	cmd           string
+	args          []string
+	executionName string
 }
 
-func NewTaskRunner(cmd string, args []string, sfnapi sfniface.SFNAPI, taskInput string, taskToken string) TaskRunner {
-	// if the task input is an array of strings, interpret these as an args array
-	// otherwise pass the raw input as a single arg
-	var taskInputArgs []string
-	if err := json.Unmarshal([]byte(taskInput), &taskInputArgs); err != nil {
-		taskInputArgs = []string{taskInput}
+func NewTaskRunner(
+	cmd string, args []string, sfnapi sfniface.SFNAPI, input, taskToken string,
+) (TaskRunner, error) {
+	var taskInput map[string]interface{}
+	if err := json.Unmarshal([]byte(input), &taskInput); err != nil {
+		return TaskRunner{}, fmt.Errorf("Input must be a json object: %s", err)
 	}
+	executionName, _ := taskInput["_EXECUTION_NAME"].(string)
 
-	return TaskRunner{
-		sfnapi:    sfnapi,
-		taskToken: taskToken,
-		cmd:       cmd,
-		args:      append(args, taskInputArgs...),
+	marshaledJob, err := json.Marshal(taskInput)
```

Comment:
one more `Job` still used here

Reply:
Renamed in d872d9befa717f36cfe354b5bfc3c164ecc5b39d